tests/python/gpu/test_profiler_gpu.py

```diff
@@ -70,23 +74,20 @@ def test_gpu_memory_profiler_symbolic():
 
     # Sample gpu_memory_profile.csv:
     # "Attribute Name","Requested Size","Device","Actual Size","Reuse?"
+    # <unk>:_head_grad_0,16777216,0,16777216,0
     # init:_random_uniform,33554432,0,33554432,1
     # init:_random_uniform,8388608,0,8388608,1
     # resource:temp_space (sample_op.h +365),8,0,4096,0
     # symbol:arg_grad:unknown,8388608,0,8388608,0
     # symbol:arg_grad:unknown,33554432,0,33554432,0
-    # tensordot:dot,16777216,0,16777216,0
-    # tensordot:dot_backward,33554432,0,33554432,0
-    # tensordot:dot_backward,8388608,0,8388608,0
-    # tensordot:dot_head_grad,16777216,0,16777216,0
+    # tensordot:dot0,16777216,0,16777216,0
     # tensordot:in_arg:A,8388608,0,8388608,0
     # tensordot:in_arg:B,33554432,0,33554432,0
+    # tensordot:node_0_backward,33554432,0,33554432,0
+    # tensordot:node_0_backward,8388608,0,8388608,0s
```

Comment:
The NameManager name shouldn't be overwritten in the Symbolic API. I think this is a regression introduced in https://github.com/apache/incubator-mxnet/pull/18598

Your prior fix of deleting `node->attrs.name = "node_" + std::to_string(node_count_++);` may not be correct, as it results in unnamed nodes in the imperative API, but if like to fix the naming, you can look into why https://github.com/apache/incubator-mxnet/pull/18598 causes the names to be overwritten

Reply:
Sorry I do not quite understand your comments here. I have reverted the changes on 

```C++
node->attrs.name = "node_" + std::to_string(node_count_++);
```

May I know what your expectation is here?

Reply:
When you first introduced the profiler, did you have names like `tensordot:node_0_backward,8388608,0,8388608,0s` or did you have names like `tensordot:dot_backward,8388608,0,8388608,0`?

It is correct to revert the change and reintroduce `node->attrs.name = "node_" + std::to_string(node_count_++);`. My question is if we need any extra change to avoid seeing `node_0_backward`. My hypothesis is that the `node_0_backward` in symbolic API has been introduced by switching the symbolic API to use the CachedOp.

Reply:
Yes. Your hypothesis is correct.

Reply:
Would you like to fix it as part of this PR?

Reply:
The previous approach is what I have in mind, but since that one is rejected I do not think I will be fixing it as part of this PR.

Reply:
The problem is that your previous approach results in unnamed nodes in the imperative interface. It's fine to fix it later. Would you like to open a bug report about the CachedOp issue?

Reply:
So what about this: we keep the name if the node has been named at the frontend, and name it to `node_0/1/2/...` if it has not been named.

Reply:
Did you check why `Imperative::RecordOp` or `Imperative::RecordDeferredCompute` is called here in the first place? I can't think of a reason for why it is called, but if it is correct / required to be called, yes, your workaround would be correct. Otherwise we should just ensure that `Imperative::RecordOp` or `Imperative::RecordDeferredCompute` from the symbolic API.

Reply:
The reason why they are invoked is because 

```Python
# executor.py
    def forward(self, is_train=False, **kwargs):
        ...
        with autograd.record(train_mode=is_train):
            self.outputs = self._cached_op(*self._args,
                                           default_ctx=default_ctx)
```
-> 
```Python
# ndarray.py
    def __call__(self, *args, **kwargs):
        check_call(_LIB.MXInvokeCachedOp(...
```
->
```C++
int MXInvokeCachedOp(...
  ...
  op->Forward(op_shared, ndinputs, ndoutputs, ctx);
```
->
```C++
OpStatePtr CachedOp::Forward(...
  ...
  op_state = DynamicForward(...
```
->
```C++
OpStatePtr CachedOp::DynamicForward(...
  ...
  RunGraph(...
```
->
```C++
void RunGraph(...
  ...
  if (recording) {
    Imperative::Get()->RecordOp(...
  }
```

Reply:
> So what about this: we keep the name if the node has been named at the frontend (i.e. `node->attrs.name != ""`), and name it to `node_0/1/2/...` if it has not been named.

I think I will go for this solution.